src/layouts/AppMenu/AppMenu.astro

```diff
@@ -0,0 +1,113 @@
+---
+import type { AppMenuFragment } from '@lib/datocms/types';
+import { t } from '@lib/i18n';
+import type { PageUrl } from '@lib/seo';
+import AppLogo from '@components/AppLogo/AppLogo.astro';
+import IconButton from '@components/IconButton/IconButton.astro';
+import LocaleSelector from '@components/LocaleSelector/LocaleSelector.astro';
+import SearchLink from '@components/SearchLink/SearchLink.astro';
+import UnstyledList from '@components/UnstyledList/UnstyledList.astro';
+import AppMenuItem from './AppMenuItem.astro';
+
+export interface Props {
+  items: AppMenuFragment['menuItems'];
+  pageUrls: PageUrl[];
+}
+const { items, pageUrls } = Astro.props;
+---
+
+<app-menu>
+  <nav class="main-menu" aria-labelledby="main-menu-title">
+    <span id="main-menu-title" class="a11y-sr-only">{t('main_menu')}</span>
+    <UnstyledList class:list="main-menu__list" data-menu-list>
+      {items.map((item) => <AppMenuItem {item} class:list="main-menu__item" />)}
+      <li class="main-menu__item main-menu__item--last"><SearchLink /></li>
+    </UnstyledList>
+  </nav>
+  <LocaleSelector pageUrls={pageUrls} />
+  <IconButton icon="menu" label={t('open_menu')} data-menu-button hidden />
+
+  <dialog data-menu-dialog>
```

Comment:
No popover API? 😅

Reply:
No, that would open as `.show()` instead of `.showModal()`. So can't use popover API :-/